### ISSUE TITLE
forward email with srs - is this possible?

### ISSUE THREAD

**Issue by kucharskim** [OpenSMTPD/OpenSMTPD#1170]
Is it possible to setup OpenSMTPD, so I can have SRS forwarding? When email comes from any SMTP server on the planet RCPT TO <first@example.com> that email is forwarded with SRS to <second@forward-example.org> and no local unix users are setup at all. I cannot figure out this from documentation and just want to be sure, is that actually implemented.

This is on `OpenBSD 7.0-stable`, and I guess I will upgrade to `7.1` coming quarter, when it is released.

```
OpenBSD 7.0 (GENERIC.MP) #5: Mon Jan 31 09:09:02 MST 2022
    root@syspatch-70-amd64.openbsd.org:/usr/src/sys/arch/amd64/compile/GENERIC.MP
```

**Comment by ichdasich** [comment]
technically, this is rather simple, just set `srs` on your outbound and add an srs key:

```
srs key "some rather long and randomly generated secret"
...
action "outbound" relay srs
```

/srs in `man smtpd.conf` ;-)

**Comment by kucharskim** [comment]
That part is the easy part, but what I don't understand is how to define mapping, of one set of email addresses to be forwarded to another set of email addresses, with SRS. From you description is not clear how to do it.

**Comment by kucharskim** [comment]
Maybe title of my issue is not reflecting where I find it challenging to setup.

**Comment by ichdasich** [comment]
SRS is applied automatically when configured like indicated. 

You have an alias foo@example.com -> bar@other.example.com.

You receive an email from foobar@external.example.com for foo@example.com. If you have SRS statements in place, OpenSMTPd will apply the SRS rewriting when forwarding that message to bar@other.example.com.

**Comment by ndTEC** [comment]
Hi,
Did you manage to do this?

**Comment by kucharskim** [comment]
> Hi, Did you manage to do this?

No I didn't, as I didn't had time to look deeper. I would need to setup testing domain and spare smtpd(8) machine to test more, but so far I didn't had time. Setup of SRS and apply it for specific domain and specific user mapping, is something which I don't know how to do.

**Comment by ArchangeGabriel** [comment]
I agree that this is not completely intuitive, but it turns out to be “quite” simple.
Here is a minimal config file for a forward only setup:
```
pki domain.tld cert "/etc/smtpd/tls/domain.tld.crt"
pki domain.tld key "/etc/smtpd/tls/domain.tld.key"

srs key some_long_and_random_ascii_secret

# Aliases table, can be any table backend
table aliases {first@example.com=second@forward-example.org}

# Accept connections from outside
listen on <networkinterface> hostname domain.tld tls

# Send emails to the world
action "relaysrs" relay helo domain.tld srs
match from local for any action "relaysrs"
action "alias" forward-only alias <aliases>
match from any for domain prepas.science action "alias"
```
